File: Data Structures & Algorithms/longest-consecutive-sequence/submission-4.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        num_set = set(nums)
        length = 0

        for num in num_set:
            if num - 1 not in num_set:
                seq = 0
                while num + seq in num_set:
                    seq += 1
                length = max(seq, length)
        return length

File: Data Structures & Algorithms/duplicate-integer/submission-4.cpp
class Solution {
public:
    bool hasDuplicate(vector<int>& nums) {
        std::map<int, int> numCounter;

        for (int num : nums) {
            if (numCounter.contains(num)) {
                numCounter[num]++;
            } else {
                numCounter[num] = 1;
            }
        }

        for (const auto& entry : numCounter) {
            if (entry.second > 1) {
                return true;
            }
        }

        return false;
    }
};